client/src/extension.ts

```diff
@@ -35,11 +38,74 @@ export function activate(context: ExtensionContext) {
 			fileEvents: workspace.createFileSystemWatcher('**/.clientrc')
 		}
 	}
+
+	let yseopCliPath : string = vscode.workspace.getConfiguration('yseopml').get('pathToYseopCli');
+
+	const batch = vscode.commands.registerCommand('yseopml.batch', () =>
+    {
+		ExecYseopCliCommand(yseopCliPath, "batch");
+    });
 	
+	const compile = vscode.commands.registerCommand('yseopml.compile', () =>
+    {
+		ExecYseopCliCommand(yseopCliPath, "compile");
+    });
+
 	// Create the language client and start the client.
 	console.log("Yseop.vscode-yseopml − Starting Language Client");
 	let disposable = new LanguageClient('yseopml', 'Yseop Markup Language language server', serverOptions, clientOptions, true).start();
 	// Push the disposable to the context's subscriptions so that the 
 	// client can be deactivated on extension deactivation
-	context.subscriptions.push(disposable);
+	context.subscriptions.push(disposable, batch, compile);
 }
+
+/**
+ * Execute a command of Yseop CLI using the folder from which VSCode has been oppened as the KB directory.
+ * @param yseopCliPath The aboslute path to Yseop CLI executable.
+ * @param commandName The Yseop CLI subcommand to use, like ”batch” or ”test”.
+ */
+export async function ExecYseopCliCommand(yseopCliPath: string, commandName: string) {
+	const editor = vscode.window.activeTextEditor;
+	
+	if(isNullOrUndefined(editor)) {
+		return;
+	}
+	if(isNullOrUndefined(yseopCliPath) || yseopCliPath.trim().length == 0) {
+		vscode.window.showInformationMessage("Please fill the Yseop CLI path parameter before using this command.");
+		return;
+	}
+
+	var binary_path = path.resolve(yseopCliPath);
+	if(!fs.existsSync(binary_path)) {
+		vscode.window.showInformationMessage(`The path to Yseop CLI in your settings doesn't seem to exist.\n“${binary_path}”`);
+		return;
+	}
+
+	if(isNullOrUndefined(vscode.workspace) || isNullOrUndefined(vscode.workspace.workspaceFolders)) {
+		vscode.window.showInformationMessage("This command must be used in a folder.");

```

Comment:
Not trivial to understand. Maybe reuse VSCode's own terms? “must be used from within a workspace folder”?